File: src/apigateway/api/mappers.go
package api

import pb "github.com/robbdimitrov/thoughts/src/apigateway/genproto"

func mapUser(u *pb.User) user {
	return user{
		ID:        u.Id,
		Name:      u.Name,
		Username:  u.Username,
		Email:     u.Email,
		Bio:       u.Bio,
		Posts:     u.Posts,
		Likes:     u.Likes,
		Following: u.Following,
		Followers: u.Followers,
		Followed:  u.Followed,
		Created:   u.Created,
	}
}

func mapPost(p *pb.Post) post {
	return post{
		ID:       p.Id,
		UserID:   p.UserId,
		Content:  p.Content,
		Likes:    p.Likes,
		Liked:    p.Liked,
		Reposts:  p.Reposts,
		Reposted: p.Reposted,
		Created:  p.Created,
	}
}

File: src/frontend/src/app/shared/mappers.js
function mapUser(user) {
  return {
    id: user.id,
    name: user.name,
    username: user.username,
    email: user.email,
    avatar: user.avatar,
    bio: user.bio,
    posts: user.posts,
    likes: user.likes,
    following: user.following,
    followers: user.followers,
    followed: user.followed,
    created: new Date(user.created)
  };
}

function mapPost(post) {
  return {
    id: post.id,
    user_id: post.user_id,
    content: post.content,
    likes: post.likes,
    liked: post.liked,
    reposts: post.reposts,
    reposted: post.reposted,
    created: new Date(post.created)
  };
}

export default {
  mapUser,
  mapPost
};
